File: apps/auth-server/src/main/java/com/example/chat/auth/server/core/service/oauth/SocialOAuth2Service.java
package com.example.chat.auth.server.core.service.oauth;

public interface SocialOAuth2Service {
    SocialType getSocialType();

    OAuth2Data getUserInfo(OAuthRequest oauthRequest);

    String getAccessToken(OAuthRequest oauthRequest);

    OAuth2Data getUserInfo(String accessToken);
}

File: apps/auth-server/src/main/kotlin/com/example/chat/auth/server/core/service/oauth/SocialOAuth2Service.kt

package com.example.chat.auth.server.core.service.oauth

interface SocialOAuth2Service {
    val socialType: SocialType
    fun getUserInfo(oauthRequest: OAuthRequest): OAuth2Data
    fun getAccessToken(oauthRequest: OAuthRequest): String
    fun getUserInfo(accessToken: String): OAuth2Data
}
